### ISSUE TITLE
Suppress Build Warnings

### ISSUE THREAD

**Issue by Shashi456** [tensorflow/swift-apis#395]
@eaplatanios @rxwei So When i tried to build on master, I get these warnings. Is there anyway to suppress these warnings ?
```
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:43:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        maxPool2D(                                                                                                                                                                                                                                     withoutDerivative(at:                                                                                           
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:76:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        maxPool2D(input, filterSize: poolSize, strides: strides, padding: padding)                                              ^                                                                                                                       withoutDerivative(at:                                                     )                                     
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:123:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        maxPool3D(input, filterSize: poolSize, strides: strides, padding: padding)                                              ^                                                                                                                       withoutDerivative(at:                                                     )                                     
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:181:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        avgPool2D(                                                                                                              ^                                                                                                                       withoutDerivative(at:                                                                                          
 /mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:214:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        avgPool2D(input, filterSize: poolSize, strides: strides, padding: padding)                                              ^                                                                                                                       withoutDerivative(at:                                                     )
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:261:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        avgPool3D(input, filterSize: poolSize, strides: strides, padding: padding)                                              ^                                                                                                                       withoutDerivative(at:                                                     )                                     
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:303:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.mean(squeezingAxes: 1)                                                                                            ^                                                                                                                       withoutDerivative(at:       )                                                                                   
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:319:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.mean(squeezingAxes: [1, 2])                                                                                       ^                                                                                                                       withoutDerivative(at:            )                                                                              
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:335:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.mean(squeezingAxes: [1, 2, 3])                                                                                    ^                                                                                                                       withoutDerivative(at:               )                                                                           
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:354:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.max(squeezingAxes: 1)                                                                                             ^                                                                                                                       withoutDerivative(at:      )                                                                                    
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:370:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.max(squeezingAxes: [1, 2])                                                                                        ^                                                                                                                       withoutDerivative(at:           )                                                                               
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Pooling.swift:386:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.max(squeezingAxes: [1, 2, 3])                                                                                     ^                                                                                                                       withoutDerivative(at:              )                                                                            
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Upsampling.swift:37:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return upSampling.reshaped(to: [batchSize, timesteps * size, channels])                                                        ^                                                                                                                       withoutDerivative(at:                                           )
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Upsampling.swift:63:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return upSampling.reshaped(to: [batchSize, height * size, width * size, channels])                                             ^                                                                                                                       withoutDerivative(at:                                                      )                             /mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Convolutional.swift:567:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        input.padded(forSizes: [(padding.0, padding.1)])                                                                        ^                                                                                                                       withoutDerivative(at:                           )                                                               
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Convolutional.swift:599:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return input.padded(forSizes: [padding.0, padding.1])                                                                          ^                                                                                                                       withoutDerivative(at:                         )                                                          
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Convolutional.swift:631:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return input.padded(forSizes: [padding.0, padding.1, padding.2])                                                               ^                                                                                                                       withoutDerivative(at:                                    )                                              
 /mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Core.swift:82:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return input.reshaped(to: [batchSize, remaining])                                                                              ^                                                                                                                       withoutDerivative(at:                     )                                                              
/mnt/g/code/swift-apis/Sources/TensorFlow/Layers/Core.swift:116:16: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
        return input.reshaped(toShape: shape)                                                                                          ^                                                                                                                       withoutDerivative(at:         )  
```

**Comment by Shashi456** [comment]
Another warning while testing, 
```
/mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:23:9: warning: result does not depend on differentiation arguments and will always have a zero derivative; do you want to use 'withoutDerivative(at:)'?
       sigmoid(input)                                                                                                   
       ^  
       withoutDerivative(at: )
```

**Comment by eaplatanios** [comment]
@Shashi456 these warnings are due to a bug in the compiler but they should not affect results. The bug will hopefully be fixed soon.

**Comment by Shashi456** [comment]
@rxwei Was this solved? I'll close this in that case.

**Comment by rxwei** [comment]
No.

**Comment by rxwei** [comment]
Currently we need a minimal reproducer and to file a bug on JIRA.

**Comment by Shashi456** [comment]
@rxwei I could file a JIRA bug and link this PR as well, But i'm not sure about the minimal reproducer, If it's with respect to these warnings then it would include the definition of these layers? So should i go ahead?

**Comment by t-ae** [comment]
Looks it's not filed yet.
I reported instead.
https://bugs.swift.org/browse/TF-775

**Comment by Shashi456** [comment]
There's another build warning here: 
```
/mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:887:13: warning: initialization of 
immutable value 'y' was never used; consider replacing with assignment to '_' or removing it

        let y = bnLayer(x)      
        ~~~~^
```

Please note, removing the line fails the test.

**Comment by dan-zheng** [comment]
> Please note, removing the line fails the test.

This is expected, because `BatchNorm.callAsFunction` updates running mean and variance, which is significant.

Simple fix in https://github.com/tensorflow/swift-apis/pull/485: `_ = bnLayer(x)`.

**Comment by Shashi456** [comment]
@rxwei @dan-zheng we do not have any build warnings popping up right now, right? So I'll close this issue?

**Comment by Shashi456** [comment]
A new set of build warnings :
``` swift
mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:51:43: warning: 'gradient(in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level
function 'TensorFlow.gradient(at:in:)' instead
                    let 𝛁mmodel = model.gradient { model -> Tensor<Float> in
/mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:1145:36: warning: 'valueWithPullback(at:in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level function 'Swift.valueWithPullback(at:_:in:)' instead
            let (outputs, _) = rnn.valueWithPullback(at: inputs) { rnn, inputs in
                                   ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:1176:40: warning: 'valueWithPullback(at:in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level function 'Swift.valueWithPullback(at:_:in:)' instead
                let (outputs, _) = rnn.valueWithPullback(at: inputs) { rnn, inputs in
                                       ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/LayerTests.swift:1207:36: warning: 'valueWithPullback(at:in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level function 'Swift.valueWithPullback(at:_:in:)' instead
            let (outputs, _) = rnn.valueWithPullback(at: inputs) { rnn, inputs in
                                   ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/OperatorTests/BasicTests.swift:79:23: warning: 'gradient(in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level
function 'TensorFlow.gradient(at:in:)' instead
        let grads = x.gradient { a -> Tensor<Float> in
                      ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/OperatorTests/BasicTests.swift:89:23: warning: 'gradient(in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level
function 'TensorFlow.gradient(at:in:)' instead
        let grads = x.gradient { a -> Tensor<Float> in
                      ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/OperatorTests/BasicTests.swift:99:23: warning: 'gradient(in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level
function 'TensorFlow.gradient(at:in:)' instead
        let grads = x.gradient { a -> Tensor<Float> in
                      ^
/mnt/g/code/swift-apis/Tests/TensorFlowTests/OperatorTests/BasicTests.swift:109:23: warning: 'gradient(in:)' is deprecated: Method-style differential operators are deprecated and will be removed; use top-level
function 'TensorFlow.gradient(at:in:)' instead
        let grads = x.gradient { a -> Tensor<Float> in
```

**Comment by Shashi456** [comment]
We currently have no build warnings popping in this repo or swift-models, please reopen whenever needed.
